pytorch_lightning/utilities/cli.py

```diff
@@ -260,6 +338,62 @@ def instantiate_trainer(self) -> None:
         self.trainer = self.trainer_class(**self.config_init['trainer'])
 
+    def add_configure_optimizers_method_to_model(self) -> None:
+        """Adds to the model an automatically generated configure_optimizers method
+
+        If a single optimizer and optionally a scheduler argument groups are added to the parser as 'AUTOMATIC',
+        then a `configure_optimizers` method is automatically implemented in the model class.
+        """
+
+        def get_automatic(class_type: Union[Type, Tuple[Type, ...]]) -> List[str]:
+            automatic = []
+            for key, (base_class, link_to) in self.parser.optimizers_and_lr_schedulers.items():
+                if not isinstance(base_class, tuple):
+                    base_class = (base_class, )
+                if link_to == 'AUTOMATIC' and any(issubclass(c, class_type) for c in base_class):
+                    automatic.append(key)
+            return automatic
+
+        optimizers = get_automatic(Optimizer)
+        lr_schedulers = get_automatic(LRSchedulerTypeTuple)
+
+        if len(optimizers) == 0:
+            return
+
+        if len(optimizers) > 1 or len(lr_schedulers) > 1:
+            raise RuntimeError(
+                f"`{self.__class__.__name__}.add_configure_optimizers_method_to_model` expects at most one optimizer "
+                f"and one lr_scheduler to be 'AUTOMATIC', but found {optimizers+lr_schedulers}."
+            )
+
+        if is_overridden('configure_optimizers', self.model):
+            warnings.warn(
+                f"`{self.model.__class__.__name__}.configure_optimizers` will be overridden by "
+                f"`{self.__class__.__name__}.add_configure_optimizers_method_to_model`."
+            )
+
+        optimizer_class = self.parser.optimizers_and_lr_schedulers[optimizers[0]][0]
+        optimizer_init = self.config_init.get(optimizers[0], {})
+        if not isinstance(optimizer_class, tuple):
+            optimizer_init = _global_add_class_path(optimizer_class, optimizer_init)
+        lr_scheduler_init = None
+        if lr_schedulers:
+            lr_scheduler_class = self.parser.optimizers_and_lr_schedulers[lr_schedulers[0]][0]
+            lr_scheduler_init = self.config_init.get(lr_schedulers[0], {})
+            if not isinstance(lr_scheduler_class, tuple):
+                lr_scheduler_init = _global_add_class_path(lr_scheduler_class, lr_scheduler_init)
+
+        def configure_optimizers(
+            self: LightningModule
+        ) -> Union[Optimizer, Tuple[List[Optimizer], List[LRSchedulerType]]]:
+            optimizer = instantiate_class(self.parameters(), optimizer_init)
+            if not lr_scheduler_init:
+                return optimizer
+            lr_scheduler = instantiate_class(optimizer, lr_scheduler_init)
+            return [optimizer], [lr_scheduler]
+
+        self.model.configure_optimizers = MethodType(configure_optimizers, self.model)
```

Comment:
noob question: what's the benefit of `MethodType` instead of setting the attribute directly?

Reply:
In this case setting the attribute directly would not work. The call would not pass the object as the first parameter `self`. It would be just a function stored in an attribute.